core/domain/html_cleaner.py

```diff
@@ -191,3 +192,371 @@ def get_rte_components(html_string: str) -> List[ComponentsDict]:
+    html_data: str, is_tag_nested_inside_tabs_or_collapsible: bool = False
+) -> None:
+    """Validate all the RTE tags.
+
+    Args:
+        html_data: str. The RTE content of the state.
+        is_tag_nested_inside_tabs_or_collapsible: bool. True when we
+            validate tags inside `Tabs` or `Collapsible` tag.
+
+    Raises:
+        ValidationError. Image does not have alt-with-value attribute.
+        ValidationError. Image alt-with-value attribute have less
+            than 5 characters.
+        ValidationError. Image does not have caption-with-value attribute.
+        ValidationError. Image caption-with-value attribute have more
+            than 500 characters.
+        ValidationError. Image does not have filepath-with-value attribute.
+        ValidationError. Image filepath-with-value attribute should not be
+            empty.
+        ValidationError. SkillReview does not have text-with-value attribute.
+        ValidationError. SkillReview text-with-value attribute should not be
+            empty.
+        ValidationError. SkillReview does not have skill_id-with-value
+            attribute.
+        ValidationError. SkillReview skill_id-with-value attribute should not be
+            empty.
+        ValidationError. Video does not have start-with-value attribute.
+        ValidationError. Video start-with-value attribute should not be empty.
+        ValidationError. Video does not have end-with-value attribute.
+        ValidationError. Video end-with-value attribute should not be empty.
+        ValidationError. Start value is greater than end value.
+        ValidationError. Video does not have autoplay-with-value attribute.
+        ValidationError. Video autoplay-with-value attribute should be boolean.
+        ValidationError. Video does not have video_id-with-value attribute.
+        ValidationError. Link does not have text-with-value attribute.
+        ValidationError. Link does not have url-with-value attribute.
+        ValidationError. Link url-with-value attribute should not be empty.
+        ValidationError. Math does not have math_content-with-value attribute.
+        ValidationError. Math math_content-with-value attribute should not be
+            empty.
+        ValidationError. Math does not have raw_latex-with-value attribute.
+        ValidationError. Math raw_latex-with-value attribute should not be
+            empty.
+        ValidationError. Math does not have svg_filename-with-value attribute.
+        ValidationError. Math svg_filename-with-value attribute should not be
+            empty.
+        ValidationError. Math svg_filename attribute does not have svg
+            extension.
+        ValidationError. Tabs tag present inside another tabs or collapsible.
+        ValidationError. Collapsible tag present inside tabs or another
+            collapsible.
+    """
+    soup = bs4.BeautifulSoup(html_data, 'html.parser')
+    empty_values = ['&quot;&quot;', '', '\'\'', '\"\"']
```

Comment:
Can this be put into a constant and used everywhere instead of redefining it?

Reply:
Done